materialsLibrary/src/sky/skyMaterial.ts

```diff
@@ -298,6 +306,9 @@ export class SkyMaterial extends PushMaterial {
             this.sunPosition.x = this.distance * Math.cos(phi);
             this.sunPosition.y = this.distance * Math.sin(phi) * Math.sin(theta);
             this.sunPosition.z = this.distance * Math.sin(phi) * Math.cos(theta);
+
+            const rotation = Quaternion.FromUnitVectors(Vector3.Up(), this.up);

```

Comment:
I would recommend using a ToRef function here as we do not want to create a quaternion per frame

Reply:
Changed it to `FromUnitVectorsToRef` and made the quaternion a private property so it doesn't get created every frame.

But I have a question:
the way I understand it, the `Vector3.Up()` will create a new vector every frame as well. Same goes for the cross product in the `FromUnitVectorsToRef` method. Shouldn't we do something with them as well? If not, isn't then making the quaternion a private property a microoptimization?

Reply:
yes totally. I think (I'm on my phone now) that we should have a UpReadOnly static property

and we should have a CrossToRef

Reply:
Nice, I used the UpReadOnly (kind of feel stupid for not finding it on my own), but I'm not sure how to use the cross product to ref, since I need to set the result to Quaternion.
I guess I'll just wait till you have a proper look.